File: authentication/src/main/java/authentication/exception/ApiErrorResponse.java
package authentication.exception;

import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
public class ApiErrorResponse {
    private final String code;
    private final String message;
    private final HttpStatus httpStatus;

    private ApiErrorResponse(String code, String message, HttpStatus httpStatus) {
        this.code = code;
        this.message = message;
        this.httpStatus = httpStatus;
    }

    public static ApiErrorResponse of(String code, String message, HttpStatus httpStatus) {
        return new ApiErrorResponse(code, message, httpStatus);
    }
}

File: api/src/main/kotlin/api/exception/ApiErrorResponse.kt
package api.exception

import org.springframework.http.HttpStatus

data class ApiErrorResponse(val code: String, val message: String, val httpStatus: HttpStatus) {

    companion object {
        fun of(code: String, message: String, httpStatus: HttpStatus): ApiErrorResponse {
            return ApiErrorResponse(code, message, httpStatus)
        }
    }
}
